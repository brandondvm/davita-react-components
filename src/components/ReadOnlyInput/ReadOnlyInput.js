import React from 'react';
import PropTypes from 'prop-types';

const ReadOnlyInput = ({ value, label, header, name }) => (
  <>
    { header && (
      <div className="ready-only-input_header">
        <i className={`di ${ header.className }`} />
        <p className="ready-only-input_header_text">{ header.text }</p>
      </div>
    )}
    <div className="ready-only-input">
      {label && <label>{label}</label>}
      <input
        type="text"
        name={name}
        value={value}
        readOnly
      />
    </div>
  </>
);

ReadOnlyInput.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  label: PropTypes.string,
  header: PropTypes.object,
};

export default ReadOnlyInput;
