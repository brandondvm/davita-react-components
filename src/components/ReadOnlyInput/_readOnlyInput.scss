.ready-only-input {
  label {
    color: $onyx;
    background: $white;
    position: relative;
    top: 9px;
    left: 0;
  }

  input {
    border: 0;
    color: $lightgray;
    font-size: 15px;
    height: 45px;
    padding: 12px 0;
    width: 100%;

    &:focus {
      outline: none;
    }
  }

  &_header {
    align-items: center;
    display: flex;
    flex-direction: row;
    &_text {
      color: $lightgray;
      font-family: $bold-font;
      font-size: 12px;
      margin: 0;
    }
    i {
      font-size: 12px;
      width: 18px;
    }
  }
}
