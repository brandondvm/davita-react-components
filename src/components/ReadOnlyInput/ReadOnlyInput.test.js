import enzyme from 'enzyme';
import React from 'react';

import ReadOnlyInput from './ReadOnlyInput';

const mockProps = {
  label: 'test label',
  value: 'test value',
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<ReadOnlyInput { ...updatedProps } />);
}

describe('Input Component', () => {
  test('Component renders correctly simple empty field', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
