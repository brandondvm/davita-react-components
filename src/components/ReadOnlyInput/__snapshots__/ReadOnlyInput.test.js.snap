// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Component renders correctly simple empty field 1`] = `
<React.Fragment>
  <div
    className="ready-only-input"
  >
    <label>
      test label
    </label>
    <input
      readOnly={true}
      type="text"
      value="test value"
    />
  </div>
</React.Fragment>
`;
